### PR TITLE
Add Subscriptions tab to Channel details page

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4993
https://issues.redhat.com/browse/ODC-5062

**Description**: 
Add channel details page.
Add subscriptions list page.

**Screen shots / Gifs for design review**: 
tmp0
tmp1

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by bmignano** [comment]
@rottencandy it seems like there might be some weird alignment issues with a few of the columns – other than that small fix it looks good to me! Thanks!

**Review by sahil143 — APPROVED** [review]
/lgtm

verified changes locally

File: frontend/packages/knative-plugin/src/utils/dynamic-channel-utils.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { EventingChannelModel, EventingKafkaChannelModel, EventingIMCModel } from '../models';
+
+export const getChannelModelForKind = (kind: string) =>
+  [EventingChannelModel, EventingIMCModel, EventingKafkaChannelModel].find(
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/utils/dynamic-channel-utils.ts:4]: can we have this util to check wrt all all channels loaded dynamically , WDYT?

File: frontend/packages/knative-plugin/src/components/eventing/EventingListPage.tsx [hunk 2]
```diff
@@ -51,6 +54,16 @@ const EventingListPage: React.FC<EventingListPageProps> = ({ match }) => {
         showTitle,
       },
     },
+    {
+      href: 'subscriptions',
+      name: t('knative-plugin~Subscriptions'),
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/eventing/EventingListPage.tsx:59]: as we are doing localize for `subscriptions` can add for above fields as well

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/eventing/EventingListPage.tsx:59]: Ah sorry i see , you added in other PR :)

File: frontend/packages/knative-plugin/src/utils/dynamic-channel-utils.ts (hunk 1 above)
Comment (rottencandy) [reply frontend/packages/knative-plugin/src/utils/dynamic-channel-utils.ts]: Removed this function and used utils from `dynamic-eventsource-utils`.

**Review by sahil143 — CHANGES_REQUESTED** [review]
@rottencandy I'm getting error while clicking on the subscriptions tab
Screenshot 2020-11-16 at 6 07 20 PM

**Comment by vikram-raj** [comment]
Look like you need to rebase and update the `knative-plugin.json` file.

Screenshot 2020-11-17 at 12 55 41 PM

**Comment by rottencandy** [comment]
Rebased and updated.

File: frontend/packages/knative-plugin/src/components/eventing/subscription-list/SubscriptionRow.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      </TableData>
+      {!customData?.channel && (
+        <TableData columnID="channel" className={tableColumnClasses[4]}>
+          <ResourceLink kind={channelKindRef} name={connectedChannel.name} namespace={namespace} />
+        </TableData>
+      )}
+      <TableData columnID="subscriber" className={tableColumnClasses[5]}>
+        {subscriber.ref ? (
+          <ResourceLink
+            kind={subscriber.ref.kind}
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/eventing/subscription-list/SubscriptionRow.tsx:63]: we should use complete reference here as it's custom resource and kind `Subscription` is there, similar is the case of TriggerRow

Comment (rottencandy) [reply frontend/packages/knative-plugin/src/components/eventing/subscription-list/SubscriptionRow.tsx]: Updated.

**Comment by invincibleJai** [comment]
/approve
/lgtm

Verified the changes, works as expected

**Review by vikram-raj — COMMENTED** [review]
Thanks @rottencandy, Verified the changes. 
/lgtm

**Pull request merged**